src/directives/compose_area.ts

```diff
@@ -93,6 +98,39 @@ export default [
                     toBytes?: number,
                 } = null;
 
+                let receiverBlocked: boolean = false;
+
+                // TODO objectEquality rausbekommen
+                scope.$watch(
```

Comment:
I only need to check/watch the "blocked" flag. My approach was similar to the one in avatar.ts. But the event wasn´t fired. To check if it would work, I enabled deep checking. And it works. But i want to get rid of the workaround. Can you give me some hints @dbrgn ?

Reply:
Hm, so this code here doesn't work if you remove the ", true" argument?

Reply:
Correct

Reply:
Try this:

```diff
diff --git a/src/directives/compose_area.ts b/src/directives/compose_area.ts
index 63cb937..e9ddc94 100644
--- a/src/directives/compose_area.ts
+++ b/src/directives/compose_area.ts
@@ -98,18 +98,6 @@ export default [
                     toBytes?: number,
                 } = null;
 
-                let receiverBlocked: boolean = false;
-
-                // TODO objectEquality rausbekommen
-                scope.$watch(
-                    () => scope.receiver,
-                    (newRec: threema.Receiver, oldRec: threema.Receiver) => {
-                    if (hasValue(newRec)) {
-                        receiverBlocked = receiverService.isBlocked(newRec);
-                        chatBlocked(receiverBlocked);
-                    }
-                }, true);
-
                 function chatBlocked(blocked: boolean) {
                     if (blocked) {
                         sendTrigger.removeClass(TRIGGER_ENABLED_CSS_CLASS);
@@ -131,6 +119,19 @@ export default [
                     }
                 }
 
+                // Initialize blocking state
+                chatBlocked(receiverService.isBlocked(scope.receiver));
+
+                // Watch `isBlocked` flag for changes
+                scope.$watch(
+                    (_scope) => receiverService.isBlocked(_scope.receiver),
+                    (isBlocked: boolean, wasBlocked: boolean) => {
+                        if (isBlocked !== wasBlocked) {
+                            chatBlocked(isBlocked);
+                        }
+                    },
+                );
+
                 /**
                  * Stop propagation of click events and hold htmlElement of the emojipicker
                  */
@@ -564,7 +565,7 @@ export default [
                 function onEmojiTrigger(ev: UIEvent): void {
                     ev.stopPropagation();
                     // TODO maybe simlify
-                    if (receiverBlocked) {
+                    if (receiverService.isBlocked(scope.receiver)) {
                         hideEmojiPicker();
                         return;
                     }
@@ -663,7 +664,7 @@ export default [
                 function onFileTrigger(ev: UIEvent): void {
                     ev.preventDefault();
                     ev.stopPropagation();
-                    if (receiverBlocked) {
+                    if (receiverService.isBlocked(scope.receiver)) {
                         return;
                     }
                     const input = element[0].querySelector('.file-input') as HTMLInputElement;
@@ -673,7 +674,7 @@ export default [
                 function onSendTrigger(ev: UIEvent): boolean {
                     ev.preventDefault();
                     ev.stopPropagation();
-                    if (receiverBlocked) {
+                    if (receiverService.isBlocked(scope.receiver)) {
                         return;
                     }
                     return sendText();
```

Unfortunately in AngularJS it's quite tricky to get data binding right :confused:

Reply:
Works like a charm :) Thanks 👍 
Unfortunately i´m unable to push to my branch atm. I´ll try to fix it over the weekend.